src/library.js

```diff
@@ -3267,6 +3267,40 @@ LibraryManager.library = {
   },
   emscripten_asm_const_double: 'emscripten_asm_const_int',
 
+  $JsValStore: {
+    values: {},
```

Comment:
Possibly `values` should be an array to hint to the engine that it can optimize for integer indices (even though it's still sparse)

Reply:
I would recommend keeping this a dictionary, since we don't really have a grasp of any pattern here to say how unsparse vs sparse this is. Sparse arrays don't have well-defined performance or memory usage behavior to my knowledge, but JS objects/maps/dictionaries have very predictable memory-time complexity. (if JS spec guarantees some complexity classes for sparse arrays, then certainly can entertain that)

Reply:
Ok, keeping this as a dictionary then.